codalab/apps/teams/tests/test_organizer_teams.py

```diff
@@ -0,0 +1,177 @@
+        assert resp.status_code == 302
+        # Make sure new team was created
+        new_team = Team.objects.get(name="Test Team 1", description="Test Team")
+
+        assert self.creator in new_team.members.all()
+
+        resp = self.client.post(
+            reverse(
+                'edit_org_team',
+                kwargs={'competition_pk': self.competition.pk, 'pk': new_team.pk}
+            ),
+            {"text_members": "test@user.com", "name": "Test Team 1", "description": "Test Team"}
+        )
+
+        assert resp.status_code == 302
+        other_teams = Team.objects.all().exclude(pk=new_team.pk)
+        # Make sure we still only have one instance
+        assert not other_teams
+        # Make sure we can still grab it by the same name and description. We didnt change anything.
+        new_team = Team.objects.get(name="Test Team 1", description="Test Team")
+
+        resp = self.client.post(
+            reverse(
+                'edit_org_team',
+                kwargs={'competition_pk': self.competition.pk, 'pk': new_team.pk}
+            ),
+            {"text_members": "test@user.com", "name": "Test Team @@@", "description": "Test Team @@@"}
+        )
+
+        assert resp.status_code == 302
+        new_team = Team.objects.get(name="Test Team @@@", description="Test Team @@@")
+        # Make sure we can grab it by the new name and description.
+        # We should still have the creator as a member
+        assert self.creator in new_team.members.all()
+
+        resp = self.client.post(
+            reverse(
+                'edit_org_team',
+                kwargs={'competition_pk': self.competition.pk, 'pk': new_team.pk}
+            ),
+            {"text_members": "test@user.com, testclientuser", "name": "Test Team @@@", "description": "Test Team @@@"}
+        )
+
+        assert resp.status_code == 302
+        new_team = Team.objects.get(name="Test Team @@@", description="Test Team @@@")
+        # Make sure we can grab it by the new name and description.
+        # We should still have the creator as a member, and now a regular user
+        assert self.creator in new_team.members.all()
+        assert self.user in new_team.members.all()
+
+        # Create a new team with our creator as a member. This should remove us from the first team.
+        resp = self.client.post(reverse('create_org_team', kwargs={'competition_pk': self.competition.pk}),
+                            {"text_members": "test@user.com", "name": "@@@Test@@@", "description": "@@@Test@@@"})
+        assert resp.status_code == 302
+        new_team_to_leave_for = Team.objects.get(name="@@@Test@@@", description="@@@Test@@@")
+        # Check that our creator got removed from the first team, and added to the second.
+        assert self.creator not in new_team.members.all()
+        assert self.creator in new_team_to_leave_for.members.all()
+
+    def test_organizer_teams_csv_file_form(self):
```

Comment:
is there a test for re-uploading the same CSV?

Reply:
No there is not. We upload a similar CSV after changing one of the team names from a previous CSV.

Reply:
Yeah this is a test case we'll definitely want, re-uploading the same CSV (in case of errors) shouldn't cause problems